src/transformers/models/ijepa/modular_ijepa.py

```diff
@@ -0,0 +1,205 @@
+import math
+from typing import Optional, Union
+
+import torch
+import torch.nn as nn
+
+from transformers.models.ijepa.configuration_ijepa import IJepaConfig
+
+from ...modeling_utils import PreTrainedModel
+from ..vit.modeling_vit import (
+    ViTAttention,
+    ViTEmbeddings,
+    ViTEncoder,
+    ViTForImageClassification,
+    ViTIntermediate,
+    ViTLayer,
+    ViTModel,
+    ViTPatchEmbeddings,
+    ViTPooler,
+    ViTSdpaAttention,
+    ViTSdpaSelfAttention,
+    ViTSelfAttention,
+    ViTSelfOutput,
+)
+
+
+class IJepaEmbeddings(ViTEmbeddings):
+    def __init__(self, config: IJepaConfig, use_mask_token: bool = False) -> None:
+        super().__init__(config, use_mask_token)
+        # Remove cls_token from IJepaEmbeddings, as it is not used in the model
+        del self.cls_token
+        num_patches = self.patch_embeddings.num_patches
+        self.position_embeddings = nn.Parameter(torch.randn(1, num_patches, config.hidden_size))
+
+    def interpolate_pos_encoding(self, embeddings: torch.Tensor, height: int, width: int) -> torch.Tensor:
```

Comment:
We should not even have to right it in the modular as it should be inherited!

Reply:
It was resolved because the method was updated. The above provided an already modified method, ViT has a CLS token, so the original method is slightly different.

e.g. ViT:
```
num_patches = embeddings.shape[1] - 1
num_positions = self.position_embeddings.shape[1] - 1
```
I-JEPA:
```
num_patches = embeddings.shape[1]
num_positions = self.position_embeddings.shape[1]
```